api/cpp/include/slint-platform.h

```diff
@@ -236,6 +250,19 @@ inline void set_platform(std::unique_ptr<Platform> platform)
                 return reinterpret_cast<const Platform *>(p)->duration_since_start().count();
 #endif
             },
+            // NOTE: if size_t is not at 32 bit unsigned integer on a 32 bit platform,
+            // this may not link with rust properly.
+            [](void *p, const SharedString *text, uint8_t clipboard) {
+                reinterpret_cast<Platform *>(p)->set_clipboard_text(*text,
+                                                                    Platform::Clipboard(clipboard));
+            },
+            [](void *p, SharedString *out_text, uint8_t clipboard) {
+                auto maybe_clipboard = reinterpret_cast<Platform *>(p)->clipboard_text(
+                        Platform::Clipboard(clipboard));
```

Comment:
I think that you can use the enum in the ffi signature, instead of `uint8_t`. Then you don't need the cast :)

Reply:
I tried this out, and I found that rust gives warnings for non exhaustive enums being passed through ffi, I suppose because the other side has no way of knowing that it will expand in the future. [Here is what the change would look like.](https://github.com/the-argus/slint/commit/7210af49f4c77986c206a95f9c546f3e873d8cdb).